### ISSUE TITLE
'ignored' option badly handled in file-name-casing 

### ISSUE THREAD

**Issue by stasberkov** [palantir/tslint#4836]
### Bug Report

- __TSLint version__: 5.19
- __TypeScript version__: 3.2
- __Running TSLint via__: CLI

#### Reproduction

Try using  `file-name-casing` having `ignored` option.
TSLint will show you a warning `Warning: file-name-casing - Unexpected casing option provided: ignored`.
This is because of line https://github.com/palantir/tslint/blob/master/src/rules/fileNameCasingRule.ts#L87
since `validCasingOptions` does not contian 'ignored' option. Since line https://github.com/palantir/tslint/blob/master/src/rules/fileNameCasingRule.ts#L43 misses it.

tslint.yaml:
```
rules:
    file-name-casing: [true, {"AppComponent.ts": "ignored", ".*": "kebab-case" } ]
```
Tried using like in here https://palantir.github.io/tslint/rules/file-name-casing/ 
```
"file-name-casing": [true, {".ts": "ignored", ".tsx": "pascal-case"}]
```

#### Actual behavior
`ignored` option claimed in docs but cannot be used.

#### Expected behavior
ability to use `ignored` option.

**Comment by adidahiya** [comment]
Can you paste your exact tslint config? It's hard to tell if it's a config error or a bug in the rule.

**Comment by stasberkov** [comment]
I updated issue text.
```
rules:
    file-name-casing: [true, {"AppComponent.ts": "ignored", ".*": "kebab-case" } ]
```
Tried using like in here https://palantir.github.io/tslint/rules/file-name-casing/
```
"file-name-casing": [true, {".ts": "ignored", ".tsx": "pascal-case"}]
```

**Comment by stasberkov** [comment]
`ignore` or `ignored`?

**Comment by tanmoyopenroot** [comment]
Hi @adidahiya ,
I was looking into the code and i am not sure it might be a bug.
Lets consider the following config
```javascript
{
    "rules": {
        "file-name-casing": [
            true,
            {
                ".ts": "kebab-case",
                ".compoent.ts": "camel-case"
            }
        ]
    }
}
```

And there are two files `file-test.ts` and `file.testComponent.ts`.
`file-test.ts` will pass as it satisfy the conditions
`file.testComponent.ts` will product an error saying 'It should be in kebab-case' but it should pass the test as condition was satisfied.

But if i consider this config for the same files `file-test.ts` and `file.testComponent.ts`.
```javascript
{
    "rules": {
        "file-name-casing": [
            true,
            {
                ".compoent.ts": "camel-case",
                ".ts": "kebab-case",
            }
        ]
    }
}
```

`file-test.ts`:- will pass as it satisfy the conditions
`file.testComponent.ts`:- it will pass 

Its because of this [line](https://github.com/palantir/tslint/blob/master/src/rules/fileNameCasingRule.ts#L92)
It only checks for first matching pattern.

So do i need to write more strict regex in the config file or is it a bug in the code?

**Comment by adidahiya** [comment]
@tanmoyopenroot that issue is separate from the one reported here. you need a more strict regex.

I opened a PR for the `ignored` issue: https://github.com/palantir/tslint/pull/4848

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
